service/preset.go

```diff
@@ -86,11 +87,37 @@ func (s *TranscodingService) newPreset(r *http.Request) swagger.GizmoJSONRespons
 	if err != nil {
 		return swagger.NewErrorResponse(err)
 	}
-	presetMap.OutputOpts = input.OutputOptions
 
-	presetMap.ProviderMapping = make(map[string]string)
 	output.Results = make(map[string]newPresetOutput)
-	for _, p := range input.Providers {
+
+	// Sometimes we try to create a new preset in a new provider but we already
+	// have the PresetMap stored. We want to update the PresetMap in such cases.
+	presetMap, err = s.db.GetPresetMap(input.Preset.Name)
+	if err == db.ErrPresetMapNotFound {
```

Comment:
There's code below this point that assumes that the presetMap is not nil. (lines 136, 141, etc) So you should probably also handle the case for when another error other than `db.ErrPresetMapNotFound` happens, yes?